### PR TITLE
Client - Indexer path as function param

### LINKED ISSUE #415: Client - Indexer path should pass to indexer methods
## Description

The indexer path should read from node info instead of hardcode in the source, therefore adding another parameter is needed in following methods
- get_outputs_id (rename to get_basic_outputs)
- get_nft_outputs
- get_alias_outputs
- get_foundry_outputs
- get_outputs_from_nft_id
- get_outputs_from_alias_id
- get_outputs_from_foundry_id

Additionally, the indexer path should add to `iota_wallet_t` and update by `wallet_update_node_config`

### LINKED ISSUE #421: Client - Check and Update UTXO Indexer REST API
## Description

UTXO Indexer REST API
https://editor.swagger.io/?url=https://raw.githubusercontent.com/iotaledger/tips/indexer-api/tips/TIP-0026/indexer-rest-api.yaml

Check and update any changed REST API endpoints or implement new REST API endpoints for:
- basic outputs
- alias outputs
- foundry outputs
- nft outputs

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change

Fixes #415 
Fixes #421 

## Type of change

- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

test_outputs_id.c

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/client/api/restful/get_outputs_id.c [hunk 1]
```diff
@@ -407,76 +551,153 @@ static char *compose_api_command(outputs_query_list_t *list, char const *const a
...
-int get_outputs_id(iota_client_conf_t const *conf, outputs_query_list_t *list, res_outputs_id_t *res) {
-  if (conf == NULL || res == NULL) {
+int get_basic_outputs(iota_client_conf_t const *conf, char const *const indexer_path, outputs_query_list_t *list,
+                      res_outputs_id_t *res) {
+  if (conf == NULL || indexer_path == NULL || res == NULL) {
     // invalid parameters
     return -1;
   }
 
+  if (strcmp(indexer_path, INDEXER_API_PATH)) {
```
Comment (matjazv) [inline src/client/api/restful/get_outputs_id.c]: `strcmp` returns negative numbers also. (this comparison occurs in multiple places in commited code)
```suggestion
  if (strcmp(indexer_path, INDEXER_API_PATH) != 0) {
```

**Review by matjazv — CHANGES_REQUESTED** [review]

File: src/client/api/restful/get_outputs_id.c (hunk 1 above)
Comment (rebinmannady93) [reply src/client/api/restful/get_outputs_id.c]: Thanks Matjaz.

**Review by matjazv — APPROVED** [review]

File: tests/client/test_config.h [hunk 2]
```diff
@@ -7,6 +7,9 @@
 // enable or disable test cases use the Tangle network.
 // We don't enable it by default but enable it for a local test is recommended.
 #define TEST_TANGLE_ENABLE 0
+
+#define TEST_INDEXER_API_PATH "/indexer/v1"
```
Comment (oopsmonk) [inline tests/client/test_config.h]: remove, `INDEXER_API_PATH` should be used which is in client/constants.h

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/client/api/restful/get_outputs_id.c [hunk 3]
```diff
@@ -75,6 +91,21 @@ size_t get_outputs_query_str_len(outputs_query_list_t *list) {
         query_str_len += strlen(elm->query_item->param);
         query_str_len += 2;  // For "&" params seperator and "=" params assignment
         break;
+      case QUERY_PARAM_HAS_NATIVE_TOKENS:
+        query_str_len += strlen(OUTPUTS_QUERY_HAS_NATIVE_TOKENS_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
```
Comment (oopsmonk) [inline src/client/api/restful/get_outputs_id.c]: maybe define a local constant inside `get_outputs_query_str_len` for this magic number?
```suggestion
        query_str_len += 2;  // For "&" params seperator and "=" params assignment
```

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**